### DISCUSSION TITLE
Pagination and limits

### DISCUSSION THREAD

**Discussion by timuckun** [prest/prest#586] [API Server] [answered]
It would be nice to able to define a max number of records a user can pull and at the same time give the users the ability to request further records page at a time.

**Comment by avelino** [comment, marked as answer]
we have parameter support `?_page_size`, it works to limit how many records should be returned per page

I noticed that I didn't have it in the documentation, I just put it in - sorry

https://docs.prestd.com/query-statements/#filters-parameters-query-string---get

**Reply by timuckun** [reply]
Is there a way to set a global limit in the config file? I want to force the users to use pagination. For example i want to set a limit that no query can return more than 100 records. if the user wants records 101-200 he has to specify an offset

**Reply by avelino** [reply]
Today it is not possible to reset the default value of `defaultPageSize`, is this a real necessity?
https://github.com/prest/prest/blob/c88a907928dc5635649ecbb20943d3162668435d/adapters/postgres/postgres.go#L37

if so, we can put it on the implementation roadmap, do you want to contribute to this implementation? I am available to assist
